src/libbson/src/bson/bson-types.h

```diff
@@ -127,7 +127,7 @@ typedef struct _bson_json_opts_t bson_json_opts_t;
 BSON_ALIGNED_BEGIN (128) typedef struct _bson_t {
    uint32_t flags; /* Internal flags for the bson_t. */
    uint32_t len;   /* Length of BSON data. */
-   char *canary;   /* For valgrind check */
+   char *canary;   /* For memcheck. */
```

Comment:
Given we no longer use Valgrind, I was re-defining "memcheck" to simply mean "memory (leak) checks" rather than the Valgrind tool. Should tasks also be renamed to avoid using the now-inapplicable `memcheck` terminology?

Reply:
I vote yes. Maybe rename the tasks to `authentication-tests-lsan`?

Reply:
LSAN was merged into ASAN and is automatically enabled when supported, so for consistency with `test-asan-memcheck-mock-server` I think `authentication-tests-asan-memcheck` would probably be most appropriate. I will also reword the comment to `For leak checks.`